src/library_fs.js

```diff
@@ -400,32 +400,44 @@ FS.staticInit();` +
     // SOCKFS is completed.
     createStream: (stream, fd_start, fd_end) => {
       if (!FS.FSStream) {
-        FS.FSStream = /** @constructor */ function() {
-          this.shared = { };
+        FS.FSStream = function FSStream() {
+          this.shared = {flags : 0, position : 0};
         };
-        FS.FSStream.prototype = {
+        FS.FSStream.prototype = Object.create(Object.prototype);
+        FS.FSStream.prototype.constructor = FS.FSStream;
```

Comment:
This is beyond my JS knowledge :smile: What do these two lines do / why are they needed?

Reply:
Apparently this is the way you do inheritance before ES6:
```js
var SubClass = function(args) { /* ... */ }; // This is the constructor
SubClass.prototype = Object.create(SuperClass.prototype);
SubClass.prototype.constructor = SubClass
```
it is a bit puzzling. I think we could get by with `FS.FSStream.prototype = {}`. But we should set the constructor, otherwise the `stream.constructor` gives `Object` which isn't ideal.